src/node/interfaces.cpp

```diff
@@ -723,6 +723,11 @@ class ChainImpl : public Chain
             notifications.transactionAddedToMempool(entry.GetSharedTx(), 0 /* mempool_sequence */);
         }
     }
+    bool hasAssumedValidChain() override
+    {
+        return Assert(m_node.chainman)->IsSnapshotActive();
```

Comment:
nit: Any reason to not use the existing `chainman().IsSnapshotActive()`?

Reply:
I'm not sure I follow here - isn't wallet code supposed to be insulated from `chainman` references by way of this `Chain` interface here? Also I think we had a preference to avoid referring to "snapshots" outside of ChainstateManager to the extent possible. cc @ryanofsky

Reply:
Oh, I just meant to replace the `Assert(...)` with `chainman()`, if you retouch.